pact-tests/application-creation.pact.spec.ts

```diff
@@ -0,0 +1,106 @@
+import 'whatwg-fetch';
+import { jest } from '@jest/globals';
+import { k8sCreateResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { like, term } from '@pact-foundation/pact/src/dsl/matchers';
+import { pactWith } from 'jest-pact';
+import { createApplication } from '../src/utils/create-utils';
+import { PactUrlUtil, Kind } from './pact-test-utils';
+
+jest.mock('@openshift/dynamic-plugin-sdk-utils');
+
+const createResourceMock = k8sCreateResource as jest.Mock;
+
+pactWith({ consumer: 'HACdev', provider: 'HAS' }, (provider) => {
+  jest.setTimeout(10000);
+
+  describe('API Pact test', () => {
+    test('App is created', async () => {
+      const namespace = 'default';
+      const app = 'app-to-create';
+      const path = `${await PactUrlUtil.getPath(Kind.Application, namespace, app)}`;
+      const requestBody = {
+        apiVersion: 'appstudio.redhat.com/v1alpha1',
+        kind: 'Application',
+        metadata: {
+          name: app,
+          namespace,
+        },
+        spec: {
+          displayName: app,
+        },
+      };
+      const expectedResponse = {
+        apiVersion: 'appstudio.redhat.com/v1alpha1',
+        kind: 'Application',
+        metadata: {
+          creationTimestamp: term({
+            generate: '2022-01-21T13:36:30Z',
+            matcher:
+              '^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(.[0-9]+)?(Z|[+-](?:2[0-3]|[01][0-9]):[0-5][0-9])?$',
```

Comment:
these matchers look reusable, could you extract them to a common place?

I'm assuming this matches a date+time, it would be more clear if it had a nice name. And I wouldn't have to look at that expression so many times.